test/message_test.dart

```diff
@@ -255,88 +252,206 @@ void main() {
     });
   });
 
-  group("encoding", () {
-    test("is null without a content-type header", () {
-      expect(_createMessage().encoding, isNull);
-    });
+  group('encoding', () {
+    group('is null and content-type header is unchanged with', () {
+      group('no content-type header and', () {
+        test('no body', () {
+          var message = _createMessage();
+          expect(message.encoding, isNull);
+          expect(message.headers, isNot(contains('content-type')));
+        });
+
+        test('a plain ASCII body', () {
+          var message = _createMessage(body: "foo");
+          expect(message.encoding, isNull);
+          expect(message.headers, isNot(contains('content-type')));
+        });
+
+        test('body bytes', () {
+          var message = _createMessage(body: _utf8Bytes);
+          expect(message.encoding, isNull);
+          expect(message.headers, isNot(contains('content-type')));
+        });
+      });
 
-    test("is null without a charset parameter", () {
-      expect(_createMessage(headers: {'content-type': 'text/plain'}).encoding,
-          isNull);
+      group('an unknown content-type header and', () {
+        test('no body', () {
+          var message = _createMessage(headers: {
+            'Content-Type': 'text/plain; charset=not-a-real-charset'
+          });
+          expect(message.encoding, isNull);
+          expect(
+              message.headers,
+              containsPair(
+                  'content-type', 'text/plain; charset=not-a-real-charset'));
```

Comment:
So no errors here when decoding it will just turn into UTF8 decoding correct?

Reply:
Yes, if `encoding` is `null` and the charset is unknown or not set, `readAsString()` will default to using UTF-8.